identity-account-storage/src/types/key_location.rs

```diff
@@ -63,3 +67,30 @@ impl Debug for KeyLocation {
     f.write_fmt(format_args!("KeyLocation{}", self))
   }
 }
+
+pub trait IotaVerificationMethodExt {
+  /// Returns the [`KeyLocation`] of an [`IotaVerificationMethod`].
+  fn key_location(&self) -> crate::Result<KeyLocation>;
+}
+
+impl IotaVerificationMethodExt for IotaVerificationMethod {
+  fn key_location(&self) -> crate::Result<KeyLocation> {
+    let fragment: &str = self
+      .id()
+      .fragment()
+      .ok_or(crate::Error::DIDError(identity_did::Error::MissingIdFragment))?;
+    let method_data: &MethodData = self.data();
+
+    let key_type: KeyType = method_to_key_type(self.type_());
+
+    Ok(KeyLocation::new(key_type, fragment.to_owned(), method_data))
+  }
+}
+
+pub fn method_to_key_type(method_type: MethodType) -> KeyType {
+  match method_type {
+    MethodType::Ed25519VerificationKey2018 => KeyType::Ed25519,
+    MethodType::X25519KeyAgreementKey2019 => KeyType::X25519,
+    MethodType::MerkleKeyCollection2021 => todo!(),
+  }
+}
```

Comment:
Try avoid exporting public functions directly, this is only used in two places (outside of tests) so it's probably fine to inline. Also this wouldn't work in general for e.g. `JsonWebKey2020` which can hold any sort of key, so I don't think we can support a stable function like this.